src/synchronous.js

```diff
@@ -31,6 +35,14 @@ Promise.enableSynchronous = function () {
 
     return this._value;
   };
+
+  Promise.prototype.getState = function () {
+    if (this._state === 3 && this._value instanceof Promise) {
```

Comment:
You shouldn't need the `instanceof` check here.  The state is only set to 3 if the promise is resolved with another promise.